src/Nethermind/Nethermind.Consensus/Tracing/GethStyleTracer.cs

```diff
@@ -24,17 +26,20 @@ public class GethStyleTracer : IGethStyleTracer
         private readonly ChangeableTransactionProcessorAdapter _transactionProcessorAdapter;
         private readonly IBlockchainProcessor _processor;
         private readonly IReceiptStorage _receiptStorage;
+        private readonly IFileSystem _fileSystem;
 
         public GethStyleTracer(
             IBlockchainProcessor processor,
             IReceiptStorage receiptStorage,
             IBlockTree blockTree,
-            ChangeableTransactionProcessorAdapter transactionProcessorAdapter)
+            ChangeableTransactionProcessorAdapter transactionProcessorAdapter,
+            IFileSystem fileSystem)
```

Comment:
Would prefer instead of file system to go with `Stream`/`TextWriter` tracer. And build FileSystem on top of that.
Would allow us to stream the trace everywhere needed. Could be used as a default way of tracing, we could stream when using JSON RPC too! Either to HttpResponse Stream or to WebSocket/IPCSocket one!

Reply:
While I see where you're coming from, the actual implementation deals with multiple files, hence multiple streams. As of the time of implementation, I haven't had any better idea. Any clue about how to effectively organize that without sacrificing encapsulation?

Reply:
Maybe: Have a kind of  `Factory` that would have `GetStream` method for each potential file. Now we can have implementation that always returns same `Stream` -> serializing everything to one.